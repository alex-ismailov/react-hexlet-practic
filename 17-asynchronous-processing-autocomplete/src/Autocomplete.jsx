import axios from 'axios';
import React from 'react';

// BEGIN (write your solution here)

// END
