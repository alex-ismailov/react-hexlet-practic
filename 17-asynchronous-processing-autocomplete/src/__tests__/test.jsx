import '@testing-library/jest-dom';

import nock from 'nock';
import React from 'react';
import axios from 'axios';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { delay } from 'nanodelay';
import Autocomplete from '../Autocomplete.jsx';

const host = 'http://localhost';

beforeAll(() => {
  nock.disableNetConnect();
});

test('Autocomplete', async () => {
  render(<Autocomplete />);

  nock(host)
    .get('/countries')
    .query({ term: 'a' })
    .reply(200, ['albania', 'aljir']);

  const input = screen.getByRole('textbox');
  userEvent.type(input, 'a');
  const result1 = await screen.findByRole('list');
  expect(result1).toMatchSnapshot();

  nock(host)
    .get('/countries')
    .query({ term: 'al' })
    .reply(200, ['albania', 'aljir']);
  userEvent.type(input, 'l');

  await delay(10);

  const result2 = await screen.findByRole('list');
  expect(result2).toMatchSnapshot();

  nock(host)
    .get('/countries')
    .query({ term: 'alb' })
    .reply(200, ['albania']);
  userEvent.type(input, 'b');

  await delay(10);

  const result3 = await screen.findByRole('list');
  expect(result3).toMatchSnapshot();

  userEvent.clear(input);

  await delay(10);

  const result4 = screen.queryByRole('list');
  expect(result4).not.toBeInTheDocument();
});
