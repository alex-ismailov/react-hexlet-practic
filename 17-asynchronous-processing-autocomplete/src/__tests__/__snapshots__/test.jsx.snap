// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete 1`] = `
<ul>
  <li>
    albania
  </li>
  <li>
    aljir
  </li>
</ul>
`;

exports[`Autocomplete 2`] = `
<ul>
  <li>
    albania
  </li>
  <li>
    aljir
  </li>
</ul>
`;

exports[`Autocomplete 3`] = `
<ul>
  <li>
    albania
  </li>
  <li>
    aljir
  </li>
</ul>
`;
